app/src/main/java/com/quarkworks/apartmentgroceries/service/SyncUser.java

```diff
@@ -369,4 +369,42 @@ public RUser then(Task<JSONObject> task) throws Exception {
 
         return networkRequest.runNetworkRequest().continueWith(addSingleUserToRealm);
     }
+
+    public static Task<RUser> updateProfile(String fieldName, String fieldValue) {
+
+        SharedPreferences sharedPreferences =
+                MyApplication.getContext().getSharedPreferences(
+                        MyApplication.getContext()
+                                .getString(R.string.login_or_sign_up_session), 0);
+        final String userId = sharedPreferences.getString(JsonKeys.USER_ID, null);
+
+        Task<JSONObject>.TaskCompletionSource tcs = Task.create();
+        UrlTemplate template = UrlTemplateCreator.updateProfile(userId, fieldName, fieldValue);
+
+        Continuation<JSONObject, Task<RUser>> updatingProfile = new Continuation<JSONObject, Task<RUser>>() {
+            @Override
+            public Task<RUser> then(Task<JSONObject> task) throws Exception{
+                if (task.isFaulted()) {
+                    Exception exception = task.getError();
+                    Log.e(TAG, "Error in updateProfile", exception);
+                    throw exception;
+                }
+
+                try {
+                    String updatedAt = task.getResult().getString(RUser.JsonKeys.UPDATED_AT);
+                    if (TextUtils.isEmpty(updatedAt)) {
+                         throw new InvalidResponseException("Incorrect updating profile response");
+                    }
+
+                    return SyncUser.getById(userId);
```

Comment:
Most of the time, we would just get the user from realm and then update it directly since we know the operation was successful and we already have the data it is going to change to. There's no use to make another network call. It's fine for now though. You don't need to change it :)